clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -425,6 +481,50 @@ class GCEUtil {
     return new Tags(items: tagsList)
   }
 
+
+  static Autoscaler buildAutoscaler(String serverGroupName,
+                                    Operation migCreateOperation,
+                                    BasicGoogleDeployDescription description) {
+    description.autoscalingPolicy.with {
+      def gceAutoscalingPolicy = new AutoscalingPolicy(coolDownPeriodSec: coolDownPeriodSec,
+                                                       minNumReplicas: minNumReplicas,
+                                                       maxNumReplicas: maxNumReplicas)
+
+      if (cpuUtilization) {
+        gceAutoscalingPolicy.cpuUtilization =
+            new AutoscalingPolicyCpuUtilization(utilizationTarget: cpuUtilization.utilizationTarget)
+      }
+
+      if (loadBalancingUtilization) {
+        gceAutoscalingPolicy.loadBalancingUtilization =
+            new AutoscalingPolicyLoadBalancingUtilization(utilizationTarget: loadBalancingUtilization.utilizationTarget)
+      }
+
+      if (customMetricUtilizations) {
+        gceAutoscalingPolicy.customMetricUtilizations = customMetricUtilizations.collect {
+          new AutoscalingPolicyCustomMetricUtilization(metric: it.metric,
+                                                       utilizationTarget: it.utilizationTarget,
+                                                       utilizationTargetType: it.utilizationTargetType)
+        }
+      }
+
+      return new Autoscaler(name: serverGroupName,
+                            zone: migCreateOperation.zone,
+                            target: migCreateOperation.targetLink,
+                            autoscalingPolicy: gceAutoscalingPolicy)
+    }
+  }
+
+  static void calibrateTargetSizeWithAutoscaler(BasicGoogleDeployDescription description) {
+    description.autoscalingPolicy.with {
+      if (description.targetSize < minNumReplicas) {
```

Comment:
I've added some additional validation to ensure that the `max` is not less than the `min` (and unit tests for same). Was already verifying that they were both non-negative.